### ISSUE TITLE
[Question][Solved] Can unet_traced.pt (TORCH.JIT.TRACE) make diffusers faster? 

### ISSUE THREAD

**Issue by camenduru** [huggingface/diffusers#1741]
I saw it here 
https://huggingface.co/riffusion/riffusion-model-v1/tree/main/unet_traced
and found here
https://huggingface.co/docs/diffusers/optimization/fp16
and here
https://pytorch.org/docs/stable/generated/torch.jit.trace.html

![Screenshot 2022-12-15 203423](https://user-images.githubusercontent.com/54370274/208252869-4de7e3ad-9e75-4ca5-87b6-8e62eb3e760c.jpg)

If it is compiled version of the model it should be faster right? is this like `jax.jit` ?

we can save it but how can we use with diffusers?

```
unet_traced = torch.jit.trace(unet, inputs)
unet_traced.save("unet_traced.pt")
```

**Comment by camenduru** [comment]
I found it 🎉 https://github.com/riffusion/riffusion-inference/blob/40e1e51c6a4e2c97cf1bc1193820862a941df62a/riffusion/server.py#L89-L114

**Comment by camenduru** [comment]
I test it 🧪
T4, with unet_traced  without xformers  30/50 [00:04<00:01,  6.51it/s] 😲 
T4, with xformers  without unet_traced  30/50 [00:05<00:02,  6.38it/s] 😲

**Comment by camenduru** [comment]
hi @patrickvonplaten 👋 this is the [vm](https://huggingface.co/spaces/camenduru/riffusion-api/blob/main/Dockerfile) I tested on and I am preparing a colab

**Comment by camenduru** [comment]
update: it is working with this [pipeline](https://github.com/riffusion/riffusion-inference/blob/main/riffusion/riffusion_pipeline.py) but not StableDiffusionPipeline

hi @NouamaneTazi 👋 I am trying to run your code from
https://github.com/huggingface/diffusers/blob/d07f73003d4d077854869b8f73275657f280334c/docs/source/optimization/fp16.mdx?plain=1#L287-L321
how did you run this code and get 

traced UNet | 3.21s | x2.96
-- | -- | --

please tell me how can I run your code without `'TracedUNet' object has no attribute 'config'`

**Comment by camenduru** [comment]
Good news 🎉 it is working with custom_pipeline="interpolate_stable_diffusion"  but not that fast 😭 

51/51 [00:08<00:00, 5.72it/s] maybe I am creating `unet_traced.pt`  wrong idk 🤔 I will ask to riffusion team

colab: https://github.com/camenduru/notebooks/blob/main/camenduru's_unet_traced.ipynb

**Comment by camenduru** [comment]
xformers slower than jit with custom seed 😲😲😲
![Screenshot 2022-12-21 222138](https://user-images.githubusercontent.com/54370274/208986549-5cdf178f-322e-4902-9efd-0f0de3583246.jpg)

**Comment by camenduru** [comment]
xformers vs JIT

![Screenshot 2022-12-21 224006](https://user-images.githubusercontent.com/54370274/208990371-8ae241e0-4e5b-45b2-86c7-3519a4e3f8bf.jpg)

**Comment by camenduru** [comment]
colab T4 
![Screenshot 2022-12-21 232002](https://user-images.githubusercontent.com/54370274/208996097-5dd8da2a-998a-4114-b8c1-2c54851c052c.jpg)
